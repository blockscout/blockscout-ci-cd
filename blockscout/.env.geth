export IMAGE=blockscout/blockscout:latest
export NAMESPACE_NAME=e2e-test
export VARIANT=geth
export HTTP_URL=http://localhost:8544
export WS_URL=ws://localhost:8546
export RESOURCE_MODE=load