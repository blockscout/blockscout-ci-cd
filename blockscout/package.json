{
  "name": "blockscout-deployment",
  "version": "0.0.1",
  "main": "main.js",
  "types": "main.ts",
  "license": "Apache-2.0",
  "private": true,
  "scripts": {
    "import": "cdk8s import",
    "lint": "eslint . --ext .ts",
    "synth": "tsc && cdk8s synth",
    "deploy": "tsc && node main.js",
    "start": "npm run deploy && npm run forward",
    "stop": "kubectl delete namespace $NAMESPACE_NAME && pkill -f kubectl",
    "forward": "kubectl --namespace $NAMESPACE_NAME port-forward deployments/deployment $PORT:$PORT $PORT_PG:$PORT_PG $PORT_NETWORK_HTTP:$PORT_NETWORK_HTTP $PORT_NETWORK_WS:$PORT_NETWORK_WS &",
    "compile": "tsc",
    "watch": "tsc -w",
    "test": "jest",
    "upgrade": "npm i cdk8s@latest cdk8s-cli@latest",
    "upgrade:next": "npm i cdk8s@next cdk8s-cli@next"
  },
  "dependencies": {
    "@kubernetes/client-node": "^0.16.3",
    "cdk8s": "^2.3.22",
    "cdk8s-plus-22": "^2.0.0-rc.14",
    "constructs": "^10.1.34",
    "dotenv": "^16.0.1",
    "envalid": "^7.3.1",
    "lodash": "^4.17.21"
  },
  "devDependencies": {
    "@types/jest": "^26.0.24",
    "@types/node": "^14.18.21",
    "@typescript-eslint/eslint-plugin": "^5.27.1",
    "@typescript-eslint/parser": "^5.27.1",
    "cdk8s-cli": "^2.0.17",
    "eslint": "^8.17.0",
    "eslint-config-airbnb-base": "^15.0.0",
    "eslint-plugin-import": "^2.26.0",
    "jest": "^26.6.3",
    "ts-jest": "^26.5.6",
    "typescript": "^4.7.3"
  }
}
