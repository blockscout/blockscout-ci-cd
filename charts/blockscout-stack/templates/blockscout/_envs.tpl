{{- define "app_env" }}
{{- range $key, $value := .Values.blockscout.environment }}
{{- $item := get $.Values.blockscout.environment $key }}
{{- if or (kindIs "string" $item) (kindIs "int64" $item) (kindIs "bool" $item)}}
- name: {{ $key }}
  value: {{ $value | quote }}
{{- else }}
- name: {{ $key }}
  value: {{ pluck $.Values.global.env $item | first | default $item._default | quote }}
{{- end }}
{{- end }}
{{- end }}

{{- define "frontend_env" }}
{{- range $key, $value := .Values.frontend.environment }}
{{- $item := get $.Values.frontend.environment $key }}
{{- if or (kindIs "string" $item) (kindIs "int64" $item) (kindIs "bool" $item)}}
- name: {{ $key }}
  value: {{ $value | quote }}
{{- else }}
- name: {{ $key }}
  value: {{ pluck $.Values.global.env $item | first | default $item._default | quote }}
{{- end }}
{{- end }}
{{- end }}
