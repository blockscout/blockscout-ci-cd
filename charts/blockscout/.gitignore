values-sokol-testnet.yaml
values-optimism.yaml
