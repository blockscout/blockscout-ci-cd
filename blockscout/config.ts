/* eslint-disable import/prefer-default-export */
import { cleanEnv, str } from 'envalid'
import { ResourceMode } from './pkg/blockscout'

require(`dotenv`).config()

export const env = cleanEnv(process.env, {
    IMAGE: str(),
    NAMESPACE_NAME: str(),
    WALLET: str(),
    HTTP_URL: str(),
    WS_URL: str({}),
    VARIANT: str({ choices: [`geth`, `parity`] }),
    RESOURCE_MODE: str({ choices: Object.values(ResourceMode) }),
})
