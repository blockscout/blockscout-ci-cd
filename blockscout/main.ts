import { App } from 'cdk8s'
import { K8sClient } from './client/client'
import { BlockscoutChart, ResourceMode } from './pkg/blockscout'

const app = new App()
const chart = new BlockscoutChart(
    app,
    `blockscout`,
    {},
    {
        // TODO: do spread with a prefix for env vars
        image: process.env.IMAGE || `blockscout/blockscout:latest`,
        namespaceName: process.env.NAMESPACE_NAME || `e2e-test`,
        wallet: process.env.WALLET || ``,
        httpURL: process.env.HTTP_URL || `http://localhost:8544`,
        wsURL: process.env.WS_URL || `ws://localhost:8546`,
        httpTraceURL: process.env.HTTP_URL || `http://localhost:8544`,
        variant: process.env.VARIANT || `geth`,
        coin: process.env.COIN || `DAI`,
        resourceMode: process.env.RESOURCE_MODE || ResourceMode.E2E,
        // mainnet vars
        firstBlock: process.env.FIRST_BLOCK,
        lastBlock: process.env.LAST_BLOCK,
        port: Number(process.env.PORT) || 4000,
        auth0Domain: process.env.AUTH0_DOMAIN,
        auth0clientID: process.env.AUTH0_CLIENT_ID,
        auth0clientSecret: process.env.AUTH0_CLIENT_SECRET,
        auth0CallbackURL: process.env.AUTH0_CALLBACK_URL,
        auth0LogoutReturnURL: process.env.AUTH0_LOGOUT_RETURN_URL,
        auth0LogoutURL: process.env.AUTH0_LOGOUT_URL,
        sendGridAPIKey: process.env.SENDGRID_API_KEY,
        sendGridSender: process.env.SENDGRID_SENDER,
        sendGridTemplate: process.env.SENDGRID_TEMPLATE,
        publicTagsAirtable: process.env.PUBLIC_TAGS_AIRTABLE_URL,
        publicTagsAirtableAPIKey: process.env.PUBLIC_TAGS_AIRTABLE_API_KEY,
        socketRoot: process.env.SOCKET_ROOT,
        networkPath: process.env.NETWORK_PATH,
        secretKeyBase: process.env.SECRET_KEY_BASE,
    },
)
app.synth()

new K8sClient(chart.namespaceName, chart.readySelector).deploy()
