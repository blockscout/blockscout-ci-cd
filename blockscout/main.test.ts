/* eslint-disable no-undef */
import { Testing } from 'cdk8s'
import { BlockscoutChart, ResourceMode } from './pkg/blockscout'

describe(`E2E mode`, () => {
    test(`E2E mode`, () => {
        const app = Testing.app()
        const chart = new BlockscoutChart(app, `test`, {}, {
            image: ``,
            namespaceName: ``,
            wallet: ``,
            wsURL: ``,
            httpURL: ``,
            variant: `geth`,
            port: 80,
            resourceMode: ResourceMode.E2E,
        })
        const results = Testing.synth(chart)
        expect(results).toMatchSnapshot()
    })

    test(`Load mode`, () => {
        const app = Testing.app()
        const chart = new BlockscoutChart(app, `test`, {}, {
            image: ``,
            namespaceName: ``,
            wallet: ``,
            wsURL: ``,
            httpURL: ``,
            variant: `geth`,
            port: 80,
            resourceMode: ResourceMode.E2E,
        })
        const results = Testing.synth(chart)
        expect(results).toMatchSnapshot()
    })

    test(`Chaos mode ( stateful set )`, () => {
        const app = Testing.app()
        const chart = new BlockscoutChart(app, `test`, {}, {
            image: ``,
            namespaceName: ``,
            wallet: ``,
            wsURL: ``,
            httpURL: ``,
            variant: `geth`,
            port: 80,
            resourceMode: ResourceMode.E2E,
        })
        const results = Testing.synth(chart)
        expect(results).toMatchSnapshot()
    })
})
