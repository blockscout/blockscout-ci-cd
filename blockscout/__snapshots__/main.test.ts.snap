// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E mode Chaos mode ( stateful set ) 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "labels": Object {
        "type": "blockscout-e2e-test",
      },
      "name": "",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "service",
      "namespace": "",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 80,
        },
      ],
      "selector": Object {
        "app": "blockscout-e2e",
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "data": Object {
      "genesis.json": "{
                    \\"config\\": {
                      \\"chainId\\": 1337,
                      \\"homesteadBlock\\": 0,
                      \\"eip150Block\\": 0,
                      \\"eip155Block\\": 0,
                      \\"eip158Block\\": 0,
                      \\"eip160Block\\": 0,
                      \\"byzantiumBlock\\": 0,
                      \\"constantinopleBlock\\": 0,
                      \\"petersburgBlock\\": 0,
                      \\"istanbulBlock\\": 0,
                      \\"muirGlacierBlock\\": 0,
                      \\"berlinBlock\\": 0,
                      \\"londonBlock\\": 0,
                      \\"clique\\":{
                        \\"blockperiodseconds\\": 1,
                        \\"epochlength\\":30000
                      }
                    },
                    \\"nonce\\": \\"0x0000000000000042\\",
                    \\"mixhash\\": \\"0x0000000000000000000000000000000000000000000000000000000000000000\\",
                    \\"difficulty\\": \\"1\\",
                    \\"coinbase\\": \\"0x3333333333333333333333333333333333333333\\",
                    \\"parentHash\\": \\"0x0000000000000000000000000000000000000000000000000000000000000000\\",
                    \\"extraData\\": \\"0x\\",
                    \\"gasLimit\\": \\"8000000000\\",
                    \\"alloc\\": {
                        \\"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266\\": { \\"balance\\": \\"20000000000000000000000\\" },
                        \\"0x70997970C51812dc3A010C7d01b50e0d17dc79C8\\": { \\"balance\\": \\"20000000000000000000000\\" }
                    }
                  }",
      "init.sh": "#!/bin/bash
                echo \\"/root/.ethereum/keystore not found, running 'geth init'...\\"
                rm -rf /root/.ethereum/keystore
                echo ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 >> priv1.txt
                echo 59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d >> priv2.txt
                geth account import --keystore /root/.ethereum/devnet/keystore --password /root/password.txt ./priv1.txt
                geth account import --keystore /root/.ethereum/devnet/keystore --password /root/password.txt ./priv2.txt
                geth --datadir /root/.ethereum/devnet init /root/genesis.json
                echo \\"...done!\\"
            
                geth \\"$@\\"
                ",
      "password.txt": "",
    },
    "immutable": false,
    "kind": "ConfigMap",
    "metadata": Object {
      "name": "test-geth-cm-c84725b6",
      "namespace": "",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "blockscout",
      },
      "name": "deployment",
      "namespace": "",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "blockscout-e2e",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "blockscout-e2e",
          },
          "name": "pod",
        },
        "spec": Object {
          "containers": Array [
            Object {
              "args": Array [
                "--fakepow",
                "--dev",
                "--dev.period",
                "1",
                "--datadir",
                "/root/.ethereum/devnet",
                "--keystore",
                "/root/.ethereum/devnet/keystore",
                "--password",
                "/root/password.txt",
                "--unlock",
                "0",
                "--unlock",
                "1",
                "--mine",
                "--miner.threads",
                "1",
                "--miner.etherbase",
                "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
                "--ipcpath",
                "/root/geth.ipc",
                "--http",
                "--http.vhosts",
                "*",
                "--http.addr",
                "0.0.0.0",
                "--http.port=8544",
                "--http.api",
                "eth,net,web3,debug,txpool",
                "--ws",
                "--ws.origins",
                "*",
                "--ws.addr",
                "0.0.0.0",
                "--ws.port=8546",
                "--ws.api",
                "eth,net,web3,debug,txpool",
                "--graphql",
                "--graphql.corsdomain",
                "*",
                "--allow-insecure-unlock",
                "--rpc.allow-unprotected-txs",
                "--http.corsdomain",
                "*",
                "--vmdebug",
                "--networkid=1337",
                "--rpc.txfeecap",
                "0",
              ],
              "command": Array [
                "sh",
                "./root/init.sh",
              ],
              "image": "ethereum/client-go:v1.10.18",
              "name": "geth",
              "ports": Array [
                Object {
                  "containerPort": 8544,
                  "name": "http",
                },
                Object {
                  "containerPort": 8546,
                  "name": "ws",
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "1000m",
                  "memory": "2Gi",
                },
                "requests": Object {
                  "cpu": "1000m",
                  "memory": "2Gi",
                },
              },
              "volumeMounts": Array [
                Object {
                  "mountPath": "/root/init.sh",
                  "name": "test-geth-cm-c84725b6",
                  "subPath": "init.sh",
                },
                Object {
                  "mountPath": "/root/genesis.json",
                  "name": "test-geth-cm-c84725b6",
                  "subPath": "genesis.json",
                },
                Object {
                  "mountPath": "/root/password.txt",
                  "name": "test-geth-cm-c84725b6",
                  "subPath": "password.txt",
                },
              ],
            },
            Object {
              "args": Array [
                "-c",
                "mix compile && mix ecto.create && mix ecto.migrate && mix phx.server",
              ],
              "command": Array [
                "/bin/bash",
              ],
              "env": Array [
                Object {
                  "name": "MIX_ENV",
                  "value": "prod",
                },
                Object {
                  "name": "ECTO_USE_SSL",
                  "value": "'false'",
                },
                Object {
                  "name": "COIN",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_VARIANT",
                  "value": "geth",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_HTTP_URL",
                  "value": "",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_TRACE_URL",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_WS_URL",
                  "value": "",
                },
                Object {
                  "name": "DATABASE_URL",
                  "value": "postgresql://postgres:@localhost:5432/blockscout?ssl=false",
                },
              ],
              "image": "",
              "imagePullPolicy": "Always",
              "name": "node",
              "ports": Array [
                Object {
                  "containerPort": 80,
                },
              ],
              "readinessProbe": Object {
                "httpGet": Object {
                  "path": "/",
                  "port": 80,
                },
                "initialDelaySeconds": 10,
                "periodSeconds": 2,
              },
              "resources": Object {
                "limits": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
                "requests": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
              },
            },
            Object {
              "env": Array [
                Object {
                  "name": "POSTGRES_PASSWORD",
                  "value": "postgres",
                },
                Object {
                  "name": "POSTGRES_DB",
                  "value": "explorer_test",
                },
              ],
              "image": "postgres:13.6",
              "name": "postgres",
              "ports": Array [
                Object {
                  "containerPort": 5432,
                  "name": "postgres",
                },
              ],
              "readinessProbe": Object {
                "exec": Object {
                  "command": Array [
                    "pg_isready",
                    "-U",
                    "postgres",
                  ],
                },
                "initialDelaySeconds": 10,
                "periodSeconds": 2,
              },
              "resources": Object {
                "limits": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
                "requests": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
              },
            },
          ],
          "volumes": Array [
            Object {
              "configMap": Object {
                "name": "test-geth-cm-c84725b6",
              },
              "name": "test-geth-cm-c84725b6",
            },
          ],
        },
      },
    },
  },
]
`;

exports[`E2E mode E2E mode 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "labels": Object {
        "type": "blockscout-e2e-test",
      },
      "name": "",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "service",
      "namespace": "",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 80,
        },
      ],
      "selector": Object {
        "app": "blockscout-e2e",
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "data": Object {
      "genesis.json": "{
                    \\"config\\": {
                      \\"chainId\\": 1337,
                      \\"homesteadBlock\\": 0,
                      \\"eip150Block\\": 0,
                      \\"eip155Block\\": 0,
                      \\"eip158Block\\": 0,
                      \\"eip160Block\\": 0,
                      \\"byzantiumBlock\\": 0,
                      \\"constantinopleBlock\\": 0,
                      \\"petersburgBlock\\": 0,
                      \\"istanbulBlock\\": 0,
                      \\"muirGlacierBlock\\": 0,
                      \\"berlinBlock\\": 0,
                      \\"londonBlock\\": 0,
                      \\"clique\\":{
                        \\"blockperiodseconds\\": 1,
                        \\"epochlength\\":30000
                      }
                    },
                    \\"nonce\\": \\"0x0000000000000042\\",
                    \\"mixhash\\": \\"0x0000000000000000000000000000000000000000000000000000000000000000\\",
                    \\"difficulty\\": \\"1\\",
                    \\"coinbase\\": \\"0x3333333333333333333333333333333333333333\\",
                    \\"parentHash\\": \\"0x0000000000000000000000000000000000000000000000000000000000000000\\",
                    \\"extraData\\": \\"0x\\",
                    \\"gasLimit\\": \\"8000000000\\",
                    \\"alloc\\": {
                        \\"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266\\": { \\"balance\\": \\"20000000000000000000000\\" },
                        \\"0x70997970C51812dc3A010C7d01b50e0d17dc79C8\\": { \\"balance\\": \\"20000000000000000000000\\" }
                    }
                  }",
      "init.sh": "#!/bin/bash
                echo \\"/root/.ethereum/keystore not found, running 'geth init'...\\"
                rm -rf /root/.ethereum/keystore
                echo ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 >> priv1.txt
                echo 59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d >> priv2.txt
                geth account import --keystore /root/.ethereum/devnet/keystore --password /root/password.txt ./priv1.txt
                geth account import --keystore /root/.ethereum/devnet/keystore --password /root/password.txt ./priv2.txt
                geth --datadir /root/.ethereum/devnet init /root/genesis.json
                echo \\"...done!\\"
            
                geth \\"$@\\"
                ",
      "password.txt": "",
    },
    "immutable": false,
    "kind": "ConfigMap",
    "metadata": Object {
      "name": "test-geth-cm-c84725b6",
      "namespace": "",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "blockscout",
      },
      "name": "deployment",
      "namespace": "",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "blockscout-e2e",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "blockscout-e2e",
          },
          "name": "pod",
        },
        "spec": Object {
          "containers": Array [
            Object {
              "args": Array [
                "--fakepow",
                "--dev",
                "--dev.period",
                "1",
                "--datadir",
                "/root/.ethereum/devnet",
                "--keystore",
                "/root/.ethereum/devnet/keystore",
                "--password",
                "/root/password.txt",
                "--unlock",
                "0",
                "--unlock",
                "1",
                "--mine",
                "--miner.threads",
                "1",
                "--miner.etherbase",
                "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
                "--ipcpath",
                "/root/geth.ipc",
                "--http",
                "--http.vhosts",
                "*",
                "--http.addr",
                "0.0.0.0",
                "--http.port=8544",
                "--http.api",
                "eth,net,web3,debug,txpool",
                "--ws",
                "--ws.origins",
                "*",
                "--ws.addr",
                "0.0.0.0",
                "--ws.port=8546",
                "--ws.api",
                "eth,net,web3,debug,txpool",
                "--graphql",
                "--graphql.corsdomain",
                "*",
                "--allow-insecure-unlock",
                "--rpc.allow-unprotected-txs",
                "--http.corsdomain",
                "*",
                "--vmdebug",
                "--networkid=1337",
                "--rpc.txfeecap",
                "0",
              ],
              "command": Array [
                "sh",
                "./root/init.sh",
              ],
              "image": "ethereum/client-go:v1.10.18",
              "name": "geth",
              "ports": Array [
                Object {
                  "containerPort": 8544,
                  "name": "http",
                },
                Object {
                  "containerPort": 8546,
                  "name": "ws",
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "1000m",
                  "memory": "2Gi",
                },
                "requests": Object {
                  "cpu": "1000m",
                  "memory": "2Gi",
                },
              },
              "volumeMounts": Array [
                Object {
                  "mountPath": "/root/init.sh",
                  "name": "test-geth-cm-c84725b6",
                  "subPath": "init.sh",
                },
                Object {
                  "mountPath": "/root/genesis.json",
                  "name": "test-geth-cm-c84725b6",
                  "subPath": "genesis.json",
                },
                Object {
                  "mountPath": "/root/password.txt",
                  "name": "test-geth-cm-c84725b6",
                  "subPath": "password.txt",
                },
              ],
            },
            Object {
              "args": Array [
                "-c",
                "mix compile && mix ecto.create && mix ecto.migrate && mix phx.server",
              ],
              "command": Array [
                "/bin/bash",
              ],
              "env": Array [
                Object {
                  "name": "MIX_ENV",
                  "value": "prod",
                },
                Object {
                  "name": "ECTO_USE_SSL",
                  "value": "'false'",
                },
                Object {
                  "name": "COIN",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_VARIANT",
                  "value": "geth",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_HTTP_URL",
                  "value": "",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_TRACE_URL",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_WS_URL",
                  "value": "",
                },
                Object {
                  "name": "DATABASE_URL",
                  "value": "postgresql://postgres:@localhost:5432/blockscout?ssl=false",
                },
              ],
              "image": "",
              "imagePullPolicy": "Always",
              "name": "node",
              "ports": Array [
                Object {
                  "containerPort": 80,
                },
              ],
              "readinessProbe": Object {
                "httpGet": Object {
                  "path": "/",
                  "port": 80,
                },
                "initialDelaySeconds": 10,
                "periodSeconds": 2,
              },
              "resources": Object {
                "limits": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
                "requests": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
              },
            },
            Object {
              "env": Array [
                Object {
                  "name": "POSTGRES_PASSWORD",
                  "value": "postgres",
                },
                Object {
                  "name": "POSTGRES_DB",
                  "value": "explorer_test",
                },
              ],
              "image": "postgres:13.6",
              "name": "postgres",
              "ports": Array [
                Object {
                  "containerPort": 5432,
                  "name": "postgres",
                },
              ],
              "readinessProbe": Object {
                "exec": Object {
                  "command": Array [
                    "pg_isready",
                    "-U",
                    "postgres",
                  ],
                },
                "initialDelaySeconds": 10,
                "periodSeconds": 2,
              },
              "resources": Object {
                "limits": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
                "requests": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
              },
            },
          ],
          "volumes": Array [
            Object {
              "configMap": Object {
                "name": "test-geth-cm-c84725b6",
              },
              "name": "test-geth-cm-c84725b6",
            },
          ],
        },
      },
    },
  },
]
`;

exports[`E2E mode Load mode 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "labels": Object {
        "type": "blockscout-e2e-test",
      },
      "name": "",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "service",
      "namespace": "",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 80,
        },
      ],
      "selector": Object {
        "app": "blockscout-e2e",
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "data": Object {
      "genesis.json": "{
                    \\"config\\": {
                      \\"chainId\\": 1337,
                      \\"homesteadBlock\\": 0,
                      \\"eip150Block\\": 0,
                      \\"eip155Block\\": 0,
                      \\"eip158Block\\": 0,
                      \\"eip160Block\\": 0,
                      \\"byzantiumBlock\\": 0,
                      \\"constantinopleBlock\\": 0,
                      \\"petersburgBlock\\": 0,
                      \\"istanbulBlock\\": 0,
                      \\"muirGlacierBlock\\": 0,
                      \\"berlinBlock\\": 0,
                      \\"londonBlock\\": 0,
                      \\"clique\\":{
                        \\"blockperiodseconds\\": 1,
                        \\"epochlength\\":30000
                      }
                    },
                    \\"nonce\\": \\"0x0000000000000042\\",
                    \\"mixhash\\": \\"0x0000000000000000000000000000000000000000000000000000000000000000\\",
                    \\"difficulty\\": \\"1\\",
                    \\"coinbase\\": \\"0x3333333333333333333333333333333333333333\\",
                    \\"parentHash\\": \\"0x0000000000000000000000000000000000000000000000000000000000000000\\",
                    \\"extraData\\": \\"0x\\",
                    \\"gasLimit\\": \\"8000000000\\",
                    \\"alloc\\": {
                        \\"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266\\": { \\"balance\\": \\"20000000000000000000000\\" },
                        \\"0x70997970C51812dc3A010C7d01b50e0d17dc79C8\\": { \\"balance\\": \\"20000000000000000000000\\" }
                    }
                  }",
      "init.sh": "#!/bin/bash
                echo \\"/root/.ethereum/keystore not found, running 'geth init'...\\"
                rm -rf /root/.ethereum/keystore
                echo ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 >> priv1.txt
                echo 59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d >> priv2.txt
                geth account import --keystore /root/.ethereum/devnet/keystore --password /root/password.txt ./priv1.txt
                geth account import --keystore /root/.ethereum/devnet/keystore --password /root/password.txt ./priv2.txt
                geth --datadir /root/.ethereum/devnet init /root/genesis.json
                echo \\"...done!\\"
            
                geth \\"$@\\"
                ",
      "password.txt": "",
    },
    "immutable": false,
    "kind": "ConfigMap",
    "metadata": Object {
      "name": "test-geth-cm-c84725b6",
      "namespace": "",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "labels": Object {
        "app": "blockscout",
      },
      "name": "deployment",
      "namespace": "",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "blockscout-e2e",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "blockscout-e2e",
          },
          "name": "pod",
        },
        "spec": Object {
          "containers": Array [
            Object {
              "args": Array [
                "--fakepow",
                "--dev",
                "--dev.period",
                "1",
                "--datadir",
                "/root/.ethereum/devnet",
                "--keystore",
                "/root/.ethereum/devnet/keystore",
                "--password",
                "/root/password.txt",
                "--unlock",
                "0",
                "--unlock",
                "1",
                "--mine",
                "--miner.threads",
                "1",
                "--miner.etherbase",
                "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
                "--ipcpath",
                "/root/geth.ipc",
                "--http",
                "--http.vhosts",
                "*",
                "--http.addr",
                "0.0.0.0",
                "--http.port=8544",
                "--http.api",
                "eth,net,web3,debug,txpool",
                "--ws",
                "--ws.origins",
                "*",
                "--ws.addr",
                "0.0.0.0",
                "--ws.port=8546",
                "--ws.api",
                "eth,net,web3,debug,txpool",
                "--graphql",
                "--graphql.corsdomain",
                "*",
                "--allow-insecure-unlock",
                "--rpc.allow-unprotected-txs",
                "--http.corsdomain",
                "*",
                "--vmdebug",
                "--networkid=1337",
                "--rpc.txfeecap",
                "0",
              ],
              "command": Array [
                "sh",
                "./root/init.sh",
              ],
              "image": "ethereum/client-go:v1.10.18",
              "name": "geth",
              "ports": Array [
                Object {
                  "containerPort": 8544,
                  "name": "http",
                },
                Object {
                  "containerPort": 8546,
                  "name": "ws",
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "1000m",
                  "memory": "2Gi",
                },
                "requests": Object {
                  "cpu": "1000m",
                  "memory": "2Gi",
                },
              },
              "volumeMounts": Array [
                Object {
                  "mountPath": "/root/init.sh",
                  "name": "test-geth-cm-c84725b6",
                  "subPath": "init.sh",
                },
                Object {
                  "mountPath": "/root/genesis.json",
                  "name": "test-geth-cm-c84725b6",
                  "subPath": "genesis.json",
                },
                Object {
                  "mountPath": "/root/password.txt",
                  "name": "test-geth-cm-c84725b6",
                  "subPath": "password.txt",
                },
              ],
            },
            Object {
              "args": Array [
                "-c",
                "mix compile && mix ecto.create && mix ecto.migrate && mix phx.server",
              ],
              "command": Array [
                "/bin/bash",
              ],
              "env": Array [
                Object {
                  "name": "MIX_ENV",
                  "value": "prod",
                },
                Object {
                  "name": "ECTO_USE_SSL",
                  "value": "'false'",
                },
                Object {
                  "name": "COIN",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_VARIANT",
                  "value": "geth",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_HTTP_URL",
                  "value": "",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_TRACE_URL",
                },
                Object {
                  "name": "ETHEREUM_JSONRPC_WS_URL",
                  "value": "",
                },
                Object {
                  "name": "DATABASE_URL",
                  "value": "postgresql://postgres:@localhost:5432/blockscout?ssl=false",
                },
              ],
              "image": "",
              "imagePullPolicy": "Always",
              "name": "node",
              "ports": Array [
                Object {
                  "containerPort": 80,
                },
              ],
              "readinessProbe": Object {
                "httpGet": Object {
                  "path": "/",
                  "port": 80,
                },
                "initialDelaySeconds": 10,
                "periodSeconds": 2,
              },
              "resources": Object {
                "limits": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
                "requests": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
              },
            },
            Object {
              "env": Array [
                Object {
                  "name": "POSTGRES_PASSWORD",
                  "value": "postgres",
                },
                Object {
                  "name": "POSTGRES_DB",
                  "value": "explorer_test",
                },
              ],
              "image": "postgres:13.6",
              "name": "postgres",
              "ports": Array [
                Object {
                  "containerPort": 5432,
                  "name": "postgres",
                },
              ],
              "readinessProbe": Object {
                "exec": Object {
                  "command": Array [
                    "pg_isready",
                    "-U",
                    "postgres",
                  ],
                },
                "initialDelaySeconds": 10,
                "periodSeconds": 2,
              },
              "resources": Object {
                "limits": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
                "requests": Object {
                  "cpu": "1000m",
                  "memory": "1024Mi",
                },
              },
            },
          ],
          "volumes": Array [
            Object {
              "configMap": Object {
                "name": "test-geth-cm-c84725b6",
              },
              "name": "test-geth-cm-c84725b6",
            },
          ],
        },
      },
    },
  },
]
`;
